arche/src/troy_west/arche.clj

```diff
@@ -53,21 +40,34 @@
   [connection]
   (alia/shutdown (:session connection)))
 
+(defn encode-udt
+  [connection key value]
+  (let [encoder (get-in connection [:udt-encoders key])]
+    (encoder value)))
+
+(defn options
+  [connection key opts]
+  (or (some-> (get-in connection [:statements key :opts])
+              (merge opts))
+      opts))
+
```

Comment:
You can ```(merge opts (get-in conn...))``` even when one of those is nil, if you so please.

Reply:
👍

Reply:
yeah it might be a bit over the top but I was trying to avoid merging (and creating a new opts every execution) if unneccesary (gc pressure, stuff like that, pre-emptive optimisation)